Davinci/Assets/Davinci/Scripts/Davinci.cs

```diff
@@ -448,13 +468,13 @@ private IEnumerator ImageLoader(Texture2D texture = null)
 
         Color color;
 
-        if (targetObj != null)
+        if (targetObj != null || targetMaterial != null) { 
             switch (rendererType)
             {
                 case RendererType.renderer:
                     Renderer renderer = targetObj.GetComponent<Renderer>();
 
-                    if (renderer == null || renderer.material == null)
+                    if (renderer == null || renderer.materials[1] == null)

```

Comment:
This looks like a use-case specific change, and doesn't correspond with the use of renderer.material in the rest of the code below.